src/transforms/push.ts

```diff
@@ -169,14 +187,28 @@ export const processPush = async (github: GitHubAPI, payload) => {
 				url: repository.html_url,
 				id: repository.id,
 				commits: chunk,
-				updateSequenceId: Date.now()
+				updateSequenceId: Date.now(),
 			};
 
-			await jiraClient.devinfo.repository.update(jiraPayload);
+			const jiraResponse = await jiraClient.devinfo.repository.update(
+				jiraPayload
+			);
+			const webhookName = payload.name || "none";
+
+			if (
+				(await booleanFlag(BooleanFlags.WEBHOOK_RECEIVED_METRICS, false)) &&
+				webhookReceived
+			) {
+				calculateProcessingTimeInSeconds(
+					webhookReceived,
```

Comment:
So at this point `webhookReceived` was deserialised from the message payload JSON. Are we sure it has `Date` type?  Maybe we need to convert it to date explicitly? (I think that what could be the reason why it was failing before)

Reply:
Oh I can see it was changed to Number, so should be good now

Reply:
Changed the way I was handling this since my original PR. Originally I was using new Date() to get the currentTime: `export const getCurrentTime = () => new Date();` and the error we were receiving in the logs failed here: `webhookReceivedTime.getTime()` because it was trying to call getTime() when webhookReceivedTime didn't exist. I'm not using Date.now() and only run that block of code if `webhookReceivedTime < currentTime` so if it's a number that's higher, or any other data type, it won't run. Tests confirm this behaviour.